dll/win32/syssetup/wizard.c

```diff
@@ -448,6 +452,24 @@ DoWriteProductOption(PRODUCT_OPTION nOption)
 
     RegCloseKey(hKey);
 
+    error = RegOpenKeyExW(HKEY_LOCAL_MACHINE, s_szWindowsNT, 0, KEY_WRITE, &hKey);
+    if (error)
+        return FALSE;
+
+    /* write CSDVersion value */
+    if (nOption == PRODUCT_OPTION_WORKSTATION)
```

Comment:
The function has two places where it checks `nOption` - one switch above and this if condition. Why so complicated?

Reply:
Simplified code.